storage/storage_image.go

```diff
@@ -159,35 +162,80 @@ func (s *storageImageSource) getBlobAndLayerID(info types.BlobInfo) (rc io.ReadC
 	return rc, n, layer.ID, err
 }
 
+func (s *storageImageSource) instanceFor(instanceDigest *digest.Digest) string {
```

Comment:
(Non-blocking: This could be an ordinary function shared between source and destination.)

Reply:
Done.